aws/resource_aws_lambda_event_source_mapping.go

```diff
@@ -303,12 +317,12 @@ func resourceAwsLambdaEventSourceMappingRead(d *schema.ResourceData, meta interf
 	state := aws.StringValue(eventSourceMappingConfiguration.State)
 
 	switch state {
-	case "Enabled", "Enabling":
+	case LambdaEventSourceMappingEnabled:
 		d.Set("enabled", true)
-	case "Disabled", "Disabling":
+	case LambdaEventSourceMappingDisabled:
 		d.Set("enabled", false)
 	default:
-		log.Printf("[DEBUG] Lambda event source mapping is neither enabled nor disabled but %s", *eventSourceMappingConfiguration.State)
+		return fmt.Errorf("state is neither enabled nor disabled but %s", *eventSourceMappingConfiguration.State)
```

Comment:
Just to prevent unexpected issues for now, likely going to keep this as a warning log.